### ISSUE TITLE
Strong freeze while trying to select in the text editor

### ISSUE THREAD

**Issue by Biep** [koreader/koreader#12031]
* KOReader version:
* Device: PocketBook Era

#### Issue
I was reading and making notes in a file with the Text Editor, and at some point wanted to move a bit of text around, so I started select.  A few times the selection process did revert before being finished (e.g. after setting the end point, the selection pop-up came, showing no selection, as if I was just starting the process).
Then the system got non-responsive, and switching off and on did not work (i.e. pressing the on/off button long), and a factory reset did not work (i.e pressing it for way more than 30 seconds) - no visible reaction at all: no screen change, nor the activity light shining.
Finally I connected the device to my computer via USB, and then the underlying system came up.  49% battery, but I see "low battery" stuff in the log (as before with issue #11989).

Scary..

#### Steps to reproduce

The freeze doesn't reappear so far, but the other selection issues do:
- It seems using the keyboard arrows confounds the selection process.
- At some point I had finished it, leading to "Selection copied to clipboard", but then when trying to paste the clipboard was empty again and the Paste button greyed out.

What I do:
- Go to the PDF [this one]](https://github.com/user-attachments/files/15848480/The.Non-Existence.of.God.-.Nicholas.Everitt.pdf)
- Gesture to open the last opened text file in the TE.
- Scroll to the proper place
- Long press (and get the copy/paste pop-up)
- Get the cursor to the starting position (by tapping or by using the keyboard keys)
- Long press (sometimes I get the copy/paste pop-up here again already)
- Get the cursor to the end position (by tapping or by using the keyboard keys)
- Long press (sometimes I get the copy/paste pop-up here)
- Go to the paste location
- Long press (sometimes I get the copy/paste pop-up here).

But I just managed to select and paste fine, even twice in a row, so I have no idea what precisely triggers it.  Has the issue resolved itself?  If it happens again, I'll come back here.

##### `crash.log` (if applicable)
[crash.log](https://github.com/user-attachments/files/15848450/crash.log)

**Comment by Biep** [comment]
Oh, if you really want them, I could give you my notes (the offending text file) too - though those are a bit more private, obviously.

**Comment by mergen3107** [comment]
You can try scramble them. Most importantly symbols and other formatting will be preserved

**Comment by Biep** [comment]
Oh, I don't mind sending them in private - it is just that the ideas are unfinished, partly wrong, and so on, and I neither want to defend/discuss them nor to be seen as holding them..
How would scrambling work?

**Comment by NiLuJe** [comment]
Assuming 17:32 matches the hang time, my first suspect, as usual, would be auto-standby screwing the pooch ;).

**Comment by Biep** [comment]
Which means that I should try with different stand-by settings?  That it is a bug in KOReader concerning auto-standby?  That you want more info that I could provide (how?)?  Or yet something else?

Which reminds me: when I take KOReader out of sleep, the timer seems to start at some random position - it may take any amount of time between zero and the time set for falling asleep.

**Comment by NiLuJe** [comment]
We have little to no control over standby behavior as long as it's enabled, and the way it's currently implemented can and wxill lead to a host of weird crap.

c.f., #9493

**Comment by hius07** [comment]
Strange selecting behaviour. The algorithm is:
-long-press on any place, in the popup choose "Select", notification is shown
-move cursor to the start, long-press, notification is shown
-move cursor to the end, long-press, notification is shown
That's it, selected text is in the clipboard, no popup dialog should appear during the process.

**Comment by Biep** [comment]
You are right - I forgot to mention that in the copy/paste pop-up I choose "Select".

And indeed, the pop-up should not reappear.  It does, because somehow the system seems to forget it was selecting.

**Comment by NiLuJe** [comment]
Yup, definitely try after disabling standby ;).

**Comment by Biep** [comment]
I disabled auto-standby.  Is that the same thing?
Should I disable standby in PocketBook itself?  ("Automatic screen lock", if translated back from the Dutch)

**Comment by NiLuJe** [comment]
That's an excellent question I'm not sure I have the answer to ;).

**Comment by Biep** [comment]
> Strange selecting behaviour.  The algorithm is:
- long-press on any place, in the popup choose "Select", notification is shown
- move cursor to the start, long-press on any place, notification is shown
- move cursor to the end, long-press on any place, notification is shown

> That's it, selected text is in the clipboard, no popup dialog should appear during the process.

You are right - I forgot to mention that in the copy/paste pop-up I choose "Select".

And indeed, the pop-up should not reappear.  Ye it does, because somehow the system seems to forget it was selecting.
